### ISSUE TITLE
how to handle level progression

### ISSUE THREAD

**Issue by plankatron** [plankatron/quakemash#2]
# the problem :
 you are far into a set of missions and die, you have to start from the beginning to catch up with your coop friend who is bored and waiting for you.. 

## solutions ? 

- an entity that serves as a checkpoint spawn if it's needed
- add triggers at the end of each level to open up a portal or air tunnel by the initial spawn point ?
- allow players to create teleporters so there friends can meet them anywhere

the other option is to just force people to play all the way through, maybe the friend would choose to meet them back in the middle ? in this case : 

## possibilities : 
- trigger additional random enemies to spawn after you pass the end of a map to leave behind just incase they die
- randomly spawn in even harder enemies once a player dies to make it even harder to make progress. 
- instead of just a teleporter or an airtube have an obstacle players have to get around in order to spawn back to there friend.

**Comment by plankatron** [comment]
## from irc 
*Soopa* : tbh, I think the most expected behaviour would be respawning at the closest coop spawn near the closest living player, with a checkpoint system to not respawn you at a coop spawn further in the level progression from that player (if no one has cleared that map yet and not activated the checkpoint system)
 and one twist on this that you could do is respawning some of the nearby monsters every time a player respawns
 ammo/health economy may need some tweaks in that case, where maybe monsters that have respawned are eligable to drop pickups on death for ammo/health and etc
 basically, one player is in the lead generally and everyone else will respawn behind them, with monsters respawning in between every time a player respawns
